### PR TITLE
HW#5: Unit, Integration, E2E tests

### PR DESCRIPTION
**Pull request by OlhaRyzha**
> Added unit and integration tests with Vitest (npm run test)
> Added Playwright component tests (npm run test:ct)
> Added E2E tests with Playwright (npm run test:e2e)
> Added HTML report for E2E runs (npm run show-report)

### CODE REVIEW

File: src/tests/integration/logic/CreateTrackModal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { describe, test, expect, vi, beforeEach } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as Dialog from '@radix-ui/react-dialog';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import CreateTrackModal from '@/components/Modal/CreateTrackModal';
+import { mockExistingTrack } from '@/tests/__mocks__/mockTrack';
+
```
Comment (andrii-mos) [inline src/tests/integration/logic/CreateTrackModal.spec.tsx:9]: - Add a test that covers the error on submit (for example, if mockCreate/mockUpdate throws an error).
- Verify that the fields are validated (for example, it should not be possible to create a track with an empty name).
- Use userEvent.clear only if required by the component logic — sometimes it’s easier to test entering values “from scratch.”
- Check that the values from the track props are actually displayed in the form when editing.
- Add a test for closing the modal window without saving.

File: src/tests/__mocks__/mockTrack.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+export default 'test-file-stub';
+
```
Comment (andrii-mos) [inline src/tests/__mocks__/mockTrack.ts:2]: - Make sure mocks reflect the real API behavior (for example, all files should handle edge cases, errors, and non-standard responses)
- Separate mocks for positive and negative scenarios
- Use TypeScript types for mock objects to avoid contract mismatches

File: tests/e2e/filtersBar.spec.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
+import { test, expect, Page } from '@playwright/test';
+import { ALL_ARTISTS, ALL_GENRES } from '@/constants/labels.constant';
+import { TableState } from '@/store/slices/table/tableSlice';
+
```
[the remark this answers is not in the document]
Comment (OlhaRyzha) [reply tests/e2e/filtersBar.spec.ts:4]: fixed in this pr -https://github.com/GenesisEducationKyiv/front-end-school-3-0-OlhaRyzha/pull/31

File: src/tests/__mocks__/mockTrack.ts (hunk 2 above)
Comment (OlhaRyzha) [reply src/tests/__mocks__/mockTrack.ts:2]: fixed in this pr -https://github.com/GenesisEducationKyiv/front-end-school-3-0-OlhaRyzha/pull/31
